core/types/types.ml

```diff
@@ -1207,7 +1116,7 @@ let rec dual_type : var_map -> datatype -> datatype =
       (* Still, we might hope to find a way of preserving 'dual
          aliases' in order to simplify the pretty-printing of types... *)
       | `Alias (_, t) -> dt t
-      | t -> raise (Invalid_argument ("Attempt to dualise non-session type: " ^ show_datatype t))
+      | _t -> raise (Invalid_argument ("Attempt to dualise non-session type: " (*^  show_datatype t *)))
```

Comment:
Maybe this function should go into typeUtils since it needs to print a datatype?

Reply:
Following Daniel's comment before: Yes, that's where we should eventually collect most of the functions that are currently in `Types`. However, the dualization functions are part of a big blob of recursive functions, including the normalization code, too. That blob is then used by other stuff in `Types`, like subtyping. Thus, trying to move that particular function involves moving a big chunk of `Types` over to `TypeUtils`, which should probably not be part of this PR.

Reply:
OK, makes sense.  (I was looking at this on a phone so didn't see the context.)